docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,612 @@
+
+  dontTouch(p) // So it doesn't get constant-propped away for the example
+  dontTouch(c) // So it doesn't get constant-propped away for the example
+
+  p :#= abType.Lit(_.elements("a") -> true.B, _.elements("b") -> true.B)
+  c :#= bcType.Lit(_.elements("b") -> true.B, _.elements("c") -> true.B)
+
+  c.waive(_.elements("c")) :<>= p.waive(_.elements("a"))
+}
+```
+
+This generates the following Verilog, where `p.b` is driven from `c.b`, and `p.a`, `c.b`, and `c.c` are initialized to default values:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example10())
+```
+
+### Connecting types with optional members
+
+In the following example, we can use `:<>=` and `waive` to connect two `MyDecoupledOpts`'s, where only one has a `bits` member.
+
+```scala mdoc:silent
+class MyDecoupledOpt(hasBits: Boolean) extends Bundle {
+  val valid = Bool()
+  val ready = Flipped(Bool())
+  val bits = if (hasBits) Some(UInt(32.W)) else None
+}
+class Example6 extends RawModule {
+  val in  = IO(Flipped(new MyDecoupledOpt(true)))
+  val out = IO(new MyDecoupledOpt(false))
+  out :<>= in.waive(_.bits.get) // We can know to call .get because we can inspect in.bits.isEmpty
+}
+```
+
+This generates the following Verilog, where `ready` and `valid` are connected, and `bits` is ignored:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example6())
+```
+
+### Always ignore extra members (partial connection operator)
+
+The most unsafe connection is to connect only members that are present in both consumer and producer, and ignore all other members. This is unsafe because this connection will never error on any Chisel types.
+
+To do this, you can use `.waiveAll` and static cast to `Data`:
+
+```scala mdoc:silent
+class OnlyA extends Bundle {
+  val a = UInt(32.W)
+}
+class OnlyB extends Bundle {
+  val b = UInt(32.W)
+}
+class Example11 extends RawModule {
+  val in  = IO(Flipped(new OnlyA()))
+  val out = IO(new OnlyB())
+
+  out := DontCare
+
+  (out: Data).waiveAll :<>= (in: Data).waiveAll
+}
+```
+
+This generates the following Verilog, where nothing is connected:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example11())
+```
+
+## Techniques for connecting structurally inequivalent Chisel types
+
+`DataView` and `viewAsSupertype` creates a view of the component that has a different Chisel type. This means that a user can first create a `DataView` of the consumer or producer (or both) so that the Chisel types are structurally equivalent. This is useful when the difference between the consumer and producers aren't super nested, and also if they have rich Scala types which encode their structure. In general, `DataView` is the preferred mechanism to use (if you can) because it maintains the most about of Chisel information in the Scala type, but there are many instances where it doesn't work and thus one must fall back on `ConnectableData`.
```

Comment:
```suggestion
`DataView` and `viewAsSupertype` create a view of the component that has a different Chisel type. This means that a user can first create a `DataView` of the consumer or producer (or both) so that the Chisel types are structurally equivalent. This is useful when the difference between the consumer and producers aren't super nested, and also if they have rich Scala types which encode their structure. In general, `DataView` is the preferred mechanism to use (if you can) because it maintains the most about of Chisel information in the Scala type, but there are many instances where it doesn't work and thus one must fall back on `ConnectableData`.
```